modules/test/scripts/init-couchbase/init-buckets.sh

```diff
@@ -8,115 +8,53 @@ set -euo pipefail
-            },
-            "mapping": {
-                "default_analyzer": "standard",
-                "default_datetime_parser": "dateTimeOptional",
-                "default_field": "_all",
-                "default_mapping": {
-                    "dynamic": false,
-                    "enabled": true,
-                    "properties": {
-                        "description": {
-                            "enabled": true,
-                            "dynamic": false,
-                            "fields": [
-                                {
-                                    "docvalues": true,
-                                    "include_in_all": true,
-                                    "include_term_vectors": true,
-                                    "index": true,
-                                    "name": "description",
-                                    "store": true,
-                                    "type": "text"
-                                }
-                            ]
-                        },
-                        "type": {
-                            "enabled": true,
-                            "dynamic": false,
-                            "fields": [
-                                {
-                                    "docvalues": true,
-                                    "include_in_all": true,
-                                    "include_term_vectors": true,
-                                    "index": true,
-                                    "name": "type",
-                                    "store": true,
-                                    "type": "text"
-                                }
-                            ]
-                        }
-                    }
-                },
-                "default_type": "_default",
-                "docvalues_dynamic": true,
-                "index_dynamic": true,
-                "store_dynamic": false,
-                "type_field": "_type"
-            },
-            "store": {
-                "indexType": "scorch",
-                "segmentVersion": 15
-            }
-        },
-        "sourceType": "gocbcore",
-        "sourceName": "travel-sample",
-        "sourceParams": {},
-        "planParams": {
-            "maxPartitionsPerPIndex": 1024,
-            "indexPartitions": 1,
-            "numReplicas": 0
-        }
-}'
-
-echo "sleep 20 to allow search index to load..."
-sleep 20
\ No newline at end of file
+curl --fail -v -u ${CB_USER}:${CB_PSWD} -X PUT \
+    http://${CB_HOST}:8094/api/index/travel-sample-index \
+    -H 'cache-control: no-cache' \
+    -H 'content-type: application/json' \
+    -d @/init-couchbase/travel-sample-index.json
+
+echo "Waiting for travel-sample-index to be ready. Please wait..."
+until curl --fail -s -u ${CB_USER}:${CB_PSWD} http://${CB_HOST}:8094/api/index/hotels-index/count |
+    jq -e '.count' | grep 31591 >/dev/null; do # there are 31591 docs to be processed in this index...
```

Comment:
hardcoded number will probably bite us at some point.
Not sure if it's important, but `if [ "$num" -ge "31591" ]` might be more future proof?
Have we confirmed with the dev team that there's no other way to check the index is complete?

Reply:
In terms of the hardcoded number, I guess it's within our control as we create the search index, so it's whatever the total number of documents is for that index.

Hmm, if we check with `-ge`, what if one of the documents we need in the examples hasn't been indexed yet because we didn't wait for all to load? That's mainly why I'm checking that the number is exactly `31591`(or whatever this could change to in future).

https://docs.couchbase.com/server/7.0/rest-api/rest-fts-indexing.html#index-querying
From what I see in the REST api docs, the `count` seemed good enough, but can always ask for confirmation and improve on this later. We didn't even check this previously so we're already in a better place 😆